xbmc/utils/JobManager.cpp

```diff
@@ -164,7 +164,7 @@ void CJobQueue::CancelJobs()
 
 bool CJobQueue::IsProcessing() const
 {
-  return !m_processing.empty() || !m_jobQueue.empty();
+  return CJobManager::GetInstance().m_running && (!m_processing.empty() || !m_jobQueue.empty());
```

Comment:
@jimfcarroll  Those jobs must be executed or the app wonˋt get the corrct state. Discarding jobs is not a corect solution.

Reply:
Yeah. I thought about that too since that's what happens with the other JobQueues that are part of the JobManager. I did it this way though because 1) doing it by draining the queue seemed more involved and potentially error prone. 2) The JobManager gets turned to !m_isrunning only on shutdown. 3) If (2) were to ever change so that the state could go back to m_isrunning then the JobQueue would continue on.

TL;DR Ultimately it came down to the fact that this was trivial and foolproof.

I can work up a different solution if you'd prefer though. Just let me know.

Reply:
Yes, please do. If we discard jobs, we lose i.e. resume point on exit. Maybe other more severe things too.
There must also be some other issue with JobQueue: start video player and halt execution by setting a breakpoint somewhere in player, do some investigation for a minite, then continue and stop playback -> stop hangs because jobqueue gets never empty.

Reply:
I did that. That's how I found the problem. It wasn't a guess:

The JobQueue in the VideoPlayer isn't managed in the JobManager so when the JobManager is stopped, that particular JobQueue still has Jobs in it. THEN, on shutdown, the VideoPlayer waits for its JobQueue to cease from processing (that is, it waits until the `CJobQueue::IsProcessing()` call to return `false`). But, it never will because the JobManager is stopped but entries remain.

So this change simply makes `CJobQueue::IsProcessing` *correctly* report that once the singleton `JobManager` is no longer running, it's no longer processing.